src/octotree.js

```diff
@@ -24,9 +25,23 @@ $(document).ready(() => {
       if (event.target === window) layoutChanged();
     });
 
-    $toggler.click(toggleSidebarAndSave);
+    $pinner.click(togglePinAndSave);
+
+    $sidebar.on('keyup mouseleave mousemove', debounce({
+      callback: () => toggleSidebar(false),
+      cancelIf: ({type}) => isSidebarPinned() || type === 'mousemove',
+      deplay: DEFAULTS.SHOWTIME
+    }));
+
+    $toggler.mouseenter(() => toggleSidebar(true));
+    $toggler.click(() => toggleSidebar(false));
+
     key.filter = () => $toggler.is(':visible');
-    key(store.get(STORE.HOTKEYS), toggleSidebarAndSave);
+    key(store.get(STORE.HOTKEYS), () => {
+      togglePinAndSave();
+      toggleSidebar(isSidebarPinned());
+      isSidebarPinned() && treeView.focus();
```

Comment:
If togglePin return pin status, this could be simplify: if (togglePin()) treeView.focus();